bot/bot.py

```diff
@@ -223,81 +58,26 @@ def insert_item_into_filter_list_cache(self, item: Dict[str, str]) -> None:
+        """Cache all the data in the FilterList on the site."""
+        full_cache = await self.api_client.get('bot/filter-lists')
 
-        # Client.login() will call HTTPClient.static_login() which will create a session using
-        # this connector attribute.
-        self.http.connector = self._connector
+        for item in full_cache:
+            self.insert_item_into_filter_list_cache(item)
 
-        self.http_session = aiohttp.ClientSession(connector=self._connector)
-        self.api_client = api.APIClient(connector=self._connector)
+    async def setup_hook(self) -> None:
+        """Default Async initialisation method for Discord.py."""
+        await super().setup_hook()
 
         if self.redis_session.closed:
             # If the RedisSession was somehow closed, we try to reconnect it
             # here. Normally, this shouldn't happen.
             await self.redis_session.connect()
 
-        try:
-            await self.ping_services()
-        except Exception as e:
-            raise StartupError(e)
-
         # Build the FilterList cache
         await self.cache_filter_list_data()
 
-        await self.stats.create_socket()
-        await super().login(*args, **kwargs)
-
-    async def on_guild_available(self, guild: discord.Guild) -> None:
-        """
-        Set the internal guild available event when constants.Guild.id becomes available.
-
-        If the cache appears to still be empty (no members, no channels, or no roles), the event
-        will not be set.
-        """
-        if guild.id != constants.Guild.id:
-            return
-
-        if not guild.roles or not guild.members or not guild.channels:
-            msg = "Guild available event was dispatched but the cache appears to still be empty!"
-            log.warning(msg)
-
-            try:
-                webhook = await self.fetch_webhook(constants.Webhooks.dev_log)
-            except discord.HTTPException as e:
-                log.error(f"Failed to fetch webhook to send empty cache warning: status {e.status}")
-            else:
-                await webhook.send(f"<@&{constants.Roles.admin}> {msg}")
-
-            return
-
-        self._guild_available.set()
-
-    async def on_guild_unavailable(self, guild: discord.Guild) -> None:
-        """Clear the internal guild available event when constants.Guild.id becomes unavailable."""
-        if guild.id != constants.Guild.id:
-            return
-
-        self._guild_available.clear()
-
-    async def wait_until_guild_available(self) -> None:
-        """
-        Wait until the constants.Guild.id guild is available (and the cache is ready).
-
-        The on_ready event is inadequate because it only waits 2 seconds for a GUILD_CREATE
-        gateway event before giving up and thus not populating the cache for unavailable guilds.
-        """
-        await self._guild_available.wait()
+        scheduling.create_task(self.load_extensions(exts))
```

Comment:
Is it not awaited to avoid a deadlock with `wait_until_guild_available`? If so, I think you should leave a brief comment here.

Reply:
21acc03ca5a053881a529ed21e437060205684ac